packages/wonder-blocks-form/components/checkbox-core.js

```diff
@@ -0,0 +1,195 @@
+            hovered,
+            focused,
+            pressed,
+            ...handlers
+        } = this.props;
+
+        const stateStyles = _generateStyles(checked, error);
+
+        const defaultStyle = [
+            sharedStyles.inputReset,
+            sharedStyles.default,
+            stateStyles.default,
+            !disabled &&
+                (pressed
+                    ? stateStyles.active
+                    : (hovered || focused) && stateStyles.focus),
+            disabled && sharedStyles.disabled,
+        ];
+
+        const props = {
+            "data-test-id": testId,
+            ...handlers,
+        };
+
+        return (
+            <View style={[sharedStyles.wrapper, style]} {...props}>
+                <StyledInput
+                    type="checkbox"
+                    aria-checked={checked}
+                    checked={checked}
+                    disabled={disabled}
+                    id={id}
+                    name={groupName}
+                    // Need to specify because this is a controlled React
+                    // form component, but we handle the click via
+                    // ClickableBehavior already
+                    onChange={() => void 0}
+                    style={defaultStyle}
+                    tabIndex={-1}
+                />
+                {checked && (
+                    <Icon
+                        color={disabled ? offBlack32 : white}
+                        icon={checkboxCheck}
+                        size="small"
+                        style={sharedStyles.checkIcon}
+                    />
+                )}
+            </View>
+        );
+    }
+}
+
+const sharedStyles = StyleSheet.create({
+    wrapper: {
+        outline: "none",
+    },
+
+    // Reset the default styled input element
+    inputReset: {
+        appearance: "none",
+        WebkitAppearance: "none",
+        MozAppearance: "none",
+    },
+
+    default: {
+        height: 16,
+        width: 16,
+        margin: 0,
+        cursor: "pointer",
+        outline: "none",
+        boxSizing: "border-box",
+        borderStyle: "solid",
+        borderWidth: 1,
+        borderRadius: 3,
+    },
+
+    disabled: {
+        cursor: "auto",
+        backgroundColor: offWhite,
+        borderColor: offBlack16,
+        borderWidth: 1,
+    },
+
+    checkIcon: {
+        position: "absolute",
+        pointerEvents: "none",
+    },
+});
+
+const fadedBlue = mix(fade(blue, 0.16), white);
+const activeBlue = mix(offBlack32, blue);
+const fadedRed = mix(fade(red, 0.08), white);
+const activeRed = mix(offBlack32, red);
+
+const colors = {
+    default: {
+        faded: fadedBlue,
+        base: blue,
+        active: activeBlue,
+    },
+    error: {
+        faded: fadedRed,
+        base: red,
+        active: activeRed,
+    },
+};
+
+const styles = {};
+
+const _generateStyles = (checked, error) => {
+    // "hash" the parameters
+    const styleKey = checked + 2 * error;
```

Comment:
I don't get the `2 * error` part of this.  I thought `checked` and `error` were booleans?